src/features/postsSlice.ts

```diff
@@ -0,0 +1,34 @@
+/* eslint-disable no-param-reassign */
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { Post } from '../types/Post';
+
+export interface PostsState {
+  posts: Post[];
+  loaded: boolean;
+  hasError: boolean;
+}
+
+const initialState: PostsState = {
+  posts: [],
+  loaded: false,
+  hasError: false,
+};
+
+export const postsSlice = createSlice({
```

Comment:
you should create an asyncThunk (`createAsyncThunk` from `'@reduxjs/toolkit'`) and handle promise state there in the extra reducers